doc/cephfs/capabilities.rst

```diff
@@ -110,3 +110,70 @@ capabilities. For example:
 The 'p' represents the pin. Each capital letter corresponds to the shift
 values, and the lowercase letters after each shift are for the actual
 capabilities granted in each shift.
+
+The relation between the lock states and the capabilities
+---------------------------------------------------------
+In MDS there are four different locks for each inode, they are simplelock,
+scatterlock, filelock and locallock. Each lock has several different lock
+states, and the MDS will issue capabilities to clients based on the lock
+state.
+
+In each state the MDS Locker will always try to issue all the capabilities to the
+clients allowed, even some capabilities are not needed or wanted by the clients,
+as pre-issuing capabilities could reduce latency in some cases.
+
+If there is only one client, usually it will be the loner client for all the inodes.
+While in multiple clients case, the MDS will try to caculate a loner client out for
+each inode depending on the capabilities the clients (needed | wanted), but usually
+it will fail. The loner client will always get all the capabilities.
+
+The filelock will control files' partial metadatas' and the file contents' access
+permissions. The metadatas include **mtime**, **atime**, **size**, etc.
+
+**Fs**: Once a client has it, all other clients are denied **Fw**.
+
+**Fx**: Only the loner client is allowed this capability. Once the lock state transitions
+        to LOCK_EXCL, the loner client is granted this along with all other file capabilities
+        except the **Fl**.
+
+**Fr**: Once a client has it, the **Fb** capability will be already revoked from all the
+        other clients.
+
+        If all the clients will only request to read the same file, the lock state will be
+        transferred to LOCK_SYNC stable state finally. And all the client could have the
+        **Fscrl** capabilities from the auth MDS and **Fscr** capabilities from the
+        replica MDSes.
+
+        If there is any client wants to write the file at the same time, then the lock
+        state will be transferred to LOCK_MIX stable state finally and all the client
+        could have the **Frwl** capabilities from the auth MDS, and the **Fr** from the
+        replica MDSes. The **Fcb** capabilities will be can't have from all the clients and
+        the clients will do sync read/write.
+
+**Fw**: If there is no loner client and once a client have this capability, the **Fsxcb**
+        capabilities will be can't have to all the clients.
+
+        If there is another client want to read or write the same file, then the lock
```

Comment:
In English, "active voice" is preferred. Rather than saying "If there is another client that wants to read or write...", it's better to make the client the "actor" in the sentence and say something like "If another client wants to read or write...".